pkg/utils/transform/transform.go

```diff
@@ -48,3 +49,54 @@ func InjectNamespaceConditional(preserveNamespace string, targetNamespace string
 		return tf(u)
 	}
 }
+
+func ReplaceKind(fromKind, toKind string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+
+		if kind != fromKind {
+			return nil
+		}
+		err := unstructured.SetNestedField(u.Object, toKind, "kind")
+		if err != nil {
+			return fmt.Errorf("could change resource KIND, %q", err)
+		}
+		return nil
+	}
+}
+
+func InjectLabel(key, value string, overwrite bool, kinds ...string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+		if len(kinds) != 0 && !itemInSlice(kind, kinds) {
+			return nil
+		}
+		labels, found, err := unstructured.NestedMap(u.Object, "metadata", "labels")
+		if err != nil {
+			return fmt.Errorf("could not find labels set, %q", err)
+		}
+		if !overwrite && found {
+			if _, ok := labels[key]; ok {
+				return nil
+			}
+		}
+		if !found || labels == nil {
+			labels = map[string]interface{}{}
+		}
+		labels[key] = value
+		err = unstructured.SetNestedMap(u.Object, labels, "metadata", "labels")
```

Comment:
nit: `if err := ... ; err != nil { `

Reply:
the if statement becomes too long.
so keeping it as it is.